clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1676,29 +1678,24 @@ void FunctionVariableConstraint::handle_params
     FVConstraint *Empty = fromEmpty ? From : To;
     FVConstraint *Typed = fromEmpty ? To : From;
     auto &CS = I.getConstraints();
+    // Copy constraint variables over the empty declaration
+    for(unsigned i = 0; i < Typed->numParams(); i++) {
+      CVarSet &TypedP = Typed->getParamVar(i);
+      auto TypedV = getOnly(TypedP);
+      CVarSet EmptyP;
+      ConstraintVariable *NewV = TypedV->getCopy(CS);
+      f(TypedV, NewV);
+      EmptyP.insert(NewV);
+      Empty->paramVars.push_back(EmptyP);
+      constrainConsVarGeq(EmptyP, TypedP, CS, nullptr, Wild_to_Safe, false, &I);
```

Comment:
I think you are making a new CVar for the prototype that previously didn't have any, and then you are basically doing what would have been done before. That is, if you have code like:
```
void f(); // prototype A
// calls to f(...) here
void f(int *x); // prototype B
```
Then you are finding the CVar for B's `x` parameter and making a corresponding one for A. To merge/transplant, you then call `f` on these two parameters. The last constraint generation aims to link things up. 

But we didn't generate these constraints before, when merging, so why should we have to do it here?

I am realizing that the problem might be that you cannot use the same code to handle the above situation and this one:
```
void f(int *x);
// calls to f(...) here
void f();
```
Basically if I see the above, I can just ignore the `f()` and keep on using the original one, since it has all the parameters present. But in the first case, the first one didn't have these parameters.

We need more/better test cases to shake out these various cases.